packages/tkg-clusterclass-aws/bundle/config/upstream/base.yaml

```diff
@@ -500,6 +510,31 @@ spec:
         matchResources:
           controlPlane: true
       jsonPatches:
+      - op: add
+        path: /spec/template/spec/kubeadmConfigSpec/initConfiguration/nodeRegistration/kubeletExtraArgs/node-labels
+        valueFrom:
+          template: |
+            {{ $first := true }}
+            {{- range $key, $val := (index .TKR_DATA .builtin.controlPlane.version).labels }}
```

Comment:
Shouldn't it be part of a TKR related PR? I am confused why do you add these labels from TKR_DATA.

Reply:
This is the labels control plane already has, I copied from what `kubeadmConfigSpec/joinConfiguration` previously did. The labels I added is the next part.

Reply:
From the test output, it seems labels from TKR are duplicated.
```
{
  "beta.kubernetes.io/arch": "amd64",
  "beta.kubernetes.io/instance-type": "vsphere-vm.cpu-2.mem-8gb.os-photon",
  "beta.kubernetes.io/os": "linux",
  "image-type": "ova",
  "key1": "value1",
  "key2": "value2",
  "kubernetes.io/arch": "amd64",
  "kubernetes.io/hostname": "mgmt-97zkp-ns97s",
  "kubernetes.io/os": "linux",
  "node-role.kubernetes.io/control-plane": "",
  "node-role.kubernetes.io/master": "",
  "node.cluster.x-k8s.io/esxi-host": "10.206.189.22",
  "node.kubernetes.io/exclude-from-external-load-balancers": "",
  "node.kubernetes.io/instance-type": "vsphere-vm.cpu-2.mem-8gb.os-photon",
  "os-name": "photon",
  "os-type": "linux",
  "run.tanzu.vmware.com/tkr": "v1.23.8---vmware.2-tkg.2-zshippable"
}
````

Reply:
Will TKR labels be lost if we don't add them in `initConfiguration`?

Reply:
we don't add it if we don't add new labels, this is to fetch existing labels, then append user specified control plane labels, otherwise original labels will be overwritten by user specified control plane labels.